### DISCUSSION TITLE
Passing `max_digits` to `Query()` in path operation function causes issue

### DISCUSSION THREAD

**Discussion by amirsoroush** [fastapi/fastapi#12102] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Query

app = FastAPI()


@app.get("/")
async def router_name(count: str = Query(max_digits=9)) -> str:
    return f"{count} - hello world"
```

### Description

Hi,
I wanted to add `max_digits` parameter to the `Query()` but it doesn't work. Also the error message is irrelevant:

```none
  File "...python3.11/site-packages/fastapi/utils.py", line 101, in create_response_field
    raise fastapi.exceptions.FastAPIError(
fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that <class 'str'> is a valid Pydantic field type. If you are using a return type annotation that is not a valid Pydantic field (e.g. Union[Response, dict, None]) you can disable generating the response model from the type annotation with the path operation decorator parameter response_model=None. Read more: https://fastapi.tiangolo.com/tutorial/response-model/
```

Why does it say "*for response field!*" ? Does it have anything to do with the response type?

Why does it say "*check that `<class 'str'>` is a valid Pydantic field type*". `str` *is* indeed a valid Pydantic type.

I checked the [docstring of this parameter](https://github.com/fastapi/fastapi/blob/master/fastapi/param_functions.py#L198), it says:

```none
            """
            Maximum number of allow digits for strings.
            """
```

Seems like the problem is related to this `max_digits`, because I can pass `max_length` without any problem.

Please let me know if I'm missing something here.

### Operating System

macOS

### FastAPI Version

0.112.2

### Pydantic Version

2.8.2

### Python Version

Python 3.11.5

**Comment by meower1** [comment, marked as answer]
The max_digits parameter is not a valid argument for the Query() function in FastAPI. It's meant for fields like Decimal in Pydantic models to specify the number of digits in a numeric field, not for query parameters.

in your case you must use max_length

```py
from fastapi import FastAPI, Query

app = FastAPI()

@app.get("/")
async def router_name(count: str = Query(max_length=9)) -> str:
    return f"{count} - hello world"
```

**Reply by amirsoroush** [reply]
Thanks. True. So do you mean the doc string should be fixed? It explicitly said "for strings".

**Reply by amirsoroush** [reply]
What about the error message? Can we have more specific message, instead of "response field" ?

**Reply by meower1** [reply]
The error does seem misleading. its best if you open an issue on this matter as that would be more likely to be checked by the developers. have a good day 👋 🌷
